### DISCUSSION TITLE
Retrieving task target and verbosity level with cake frosting

### DISCUSSION THREAD

**Discussion by nathangiuliani** [cake-build/cake#3290] [Q&A]
How can I retrieve the task target with frosting? With previous releases we were able to access this via the Arguments method but leading up to 1.0.0 this is no longer presented as an argument.

It seems this might be possible using an ISetupContext in cake scripting, but I can't see how to make this work in Frosting.

As for why - mostly just a nice to have! Knowing the target allows us to only throw exceptions when the relevant parameters for a specific task are incorrect or missing.

Thanks

**Comment by patriksvensson** [comment]
@nathangiuliani You should be able to get `target` argument like always via the `Argument<string>` method. The only difference is that you will need to call the method via the `ICakeContext`.

```csharp
context.Argument<string>("target", "DefaultTaskName");
```

**Reply by nathangiuliani** [reply]
That's exactly what I'm doing - the target and verbosity arguments aren't returned.

**Reply by patriksvensson** [reply]
Sounds like a bug. Would you mind opening an issue? 👍

**Reply by augustoproiete** [reply]
Tracking on https://github.com/cake-build/cake/issues/3291
